### PR TITLE
gc: Add `--rev` and `--num` flags

### PR DESCRIPTION
**Pull request by daavoo**
Per https://github.com/iterative/dvc/pull/7971

### CODE REVIEW

File: content/docs/command-reference/gc.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ Remove unused files and directories from <abbr>cache</abbr> or [remote storage].
 ## Synopsis
 
 ```usage
-usage: dvc gc [-h] [-q | -v] [-w] [-a] [-T] [--all-commits]
+usage: dvc gc [-h] [-q | -v] [-w] [--rev <commit>] [-n <num>] [-a] [-T] [-A]
               [--all-experiments] [-c] [-r <name>] [-f] [-j <number>]
               [-p [<path> [<path> ...]]]
```
Comment (jorgeorpinel) [inline content/docs/command-reference/gc.md:10]: Why are `--rev` and `-n` before `-aTA`?

File: content/docs/command-reference/gc.md [hunk 2]
```diff
@@ -74,6 +74,13 @@ project we want to clear.
   workspace. This option is enabled automatically with the other scope options
   (below).
 
+- `--rev` - keep cached data referenced in the specified `<commit>`, as well as
+  in the workspace (implying `-w`).
```
Comment (jorgeorpinel) [inline content/docs/command-reference/gc.md:78]:
```suggestion
- `--rev <commit>` - keep cached data referenced in the specified `<commit>`, as
  well as in the workspace (implying `-w`).
```

File: content/docs/command-reference/gc.md [hunk 3]
```diff
@@ -74,6 +74,13 @@ project we want to clear.
...
+
+- `-n`, `--num` - keep cached data referenced in the last `--num` commits,
+  starting from the `--rev` `<commit>`, as well as in the workspace (implying
+  `-w`).
```
Comment (jorgeorpinel) [inline content/docs/command-reference/gc.md:82]:
```suggestion
- `-n <num>`, `--num <num>` - keep cached data referenced in the last `--num`
  commits starting from a `--rev` `<commit>` (required), as well as in the
  workspace (implying `-w`).
```

Comment (jorgeorpinel) [reply content/docs/command-reference/gc.md:82]: Is `--rev` indeed required for `-n`?

**Review by jorgeorpinel — COMMENTED** [review]
Looking good so far. Thanks @daavoo 

But there seems to be a good amount of discussion in https://github.com/iterative/dvc/pull/7971 so please request a docs update again when the design is clearer.

File: content/docs/command-reference/gc.md (hunk 3 above)
Comment (daavoo) [reply content/docs/command-reference/gc.md:82]: Yes

File: content/docs/command-reference/gc.md [hunk 4]
```diff
@@ -5,7 +5,7 @@ Remove unused files and directories from <abbr>cache</abbr> or [remote storage].
 ## Synopsis
 
 ```usage
-usage: dvc gc [-h] [-q | -v] [-w] [-a] [-T] [--all-commits]
+usage: dvc gc [-h] [-q | -v] [-w] [--rev <commit>] [-n <num>] [-a] [-T] [-A]
               [--date <YYYY-MM-DD>] [--all-experiments]
               [-c] [-r <name>] [-f] [-j <number>]
               [-p [<path> [<path> ...]]]
```
Comment (jorgeorpinel) [inline content/docs/command-reference/gc.md:11]: We need to reorg these. Maybe

```suggestion
usage: dvc gc [-h] [-q | -v]
              [-w] [--rev <commit>] [-n <num>] [-a] [-T] [-A]
              [--date <YYYY-MM-DD>] [--all-experiments]
              [-p [<path> [<path> ...]]]
              [-c] [-r <name>] [-f] [-j <number>]
```

Comment (jorgeorpinel) [reply content/docs/command-reference/gc.md:11]: Please commit ☝🏼 if you agree @daavoo cc @dberenbaum then we can merge, I think. Thanks

**Review by jorgeorpinel — APPROVED** [review]

**Pull request merged**